### ISSUE TITLE
Epic: Kickoff Orbaco STAC ipyleaflet

### ISSUE THREAD

**Issue by zacharyDez** [MAAP-Project/stac_ipyleaflet#46]
## Overview

**The MAAP platform combines data, algorithms, and computational abilities to process and share data related to NASA’s GEDI, ESA’s BIOMASS, and NASA/ISRO’s NISAR missions.** These missions generate more significant amounts of data than previous Earth observation missions. There are unique challenges to processing, storing, and sharing relevant data due to the high volume of data collected from various satellites, aircraft, and ground stations with different resolutions, coverages, and processing levels.

**MAAP aims to address unique challenges by making it easier to discover and use biomass-relevant data, integrating the data for comparison, analysis, evaluation, and generation.** An algorithm development environment (ADE) creates repeatable and sharable science tools for the research community. The software is open source and adheres to ESA’s and NASA’s commitment to open data.

**NASA and ESA are collaborating to further biomass-relevant data and metadata interoperability.** Tools have been developed to support a new approach to data stewardship. There is a data publication workflow for organizing and storing data and generating metadata to be discoverable in a cloud-based centralized location. The platform and data stewardship approaches are designed to ease barriers and promote collaboration between researchers, providers, curators, and experts across NASA and ESA.

## Key Links

- **[MAAP project documentation](https://docs.maap-project.org/en/latest/visualizing.html)**
- [STAC ipyleaflet repository](https://github.com/MAAP-Project/stac_ipyleaflet)
- [Call for support](https://docs.google.com/document/d/1DVKv3zAKjZMdhDi2TAx78sT2g6feV1GSAEJHLRtbYFM/edit#heading=h.rfi6jiigu2g3)
- [SmartSheet Project](https://rm.smartsheet.com/projects/6975782)

### Problem Statement Hypothesis

The current version of the STAC ipyleaflet project is not user-friendly and lacks clear UX flows, making it difficult for researchers to analyze large volumes of geospatial data efficiently. The project requires redesigning its user interface and experience to improve the overall user experience and streamline the analysis process. The redesign aims to create a more intuitive and user-friendly interface, with straightforward navigation and data visualization features, that can accommodate the needs of researchers working with large geospatial datasets.

### Vision

Towards a streamlined version of a geospatial-centric scientific computing notebook that empowers user customization with more of a GUI experience. The tool will focus on integrating STAC analyses and only support some other local and cloud-native data formats (GeoJSON, COG, FlatGeoBuf, GeoParquet, MVT, WCS, ZARR, HDF).  For example, a potential partnership or competition could develop with [Leafmap](https://leafmap.org/#key-features), a project DevSeed uses on many projects such as VEDA. For more details, see the [[MAAP V3 Fast Browse Proposal](https://docs.google.com/document/d/1DVKv3zAKjZMdhDi2TAx78sT2g6feV1GSAEJHLRtbYFM/edit#).

### End-Users

The MAAP platform is utilized by individuals in the scientific community who have access to the MAAP Analysis Development Environment (ADE) and can both access and generate data within the system. The platform grants users access to various external S3 buckets, some of which mandate credentials managed by the MAAP platform. Looking at the larger context, the tool should be functional in a Jupyter notebook environment for showcasing data that is either local or remote (STAC catalog) based. Remote data must only be supported in cloud-optimized formats or standard web services such as XYZ tiles and OGC Feature services.

### Stakeholders

- NASA MAAP Data Team
- NASA MAAP Platform Team (Partner)
- MAAP User Working Group (UWG)

### Success Criteria

- User Working Group satisfaction - measured with NPS
- Team satisfaction with the project - measured with NPS

### Project Landscape

Within the MAAP project, we compete indirectly with the ESA [EDAV](https://portal.val.esa-maap.org/portal-val/ESA/explore) platform.

### Risks

**Value Risks:**

- Do we understand the scale of the impact? How many users are there?
- Are there other emerging tools that aren't notebooks to be aware of concerning competing science workflows?
- How much time can we save in a science workflow through this project?

**Usability Risks:**

- Will our redesign increase our user's ability to problem solve?
- Risk or under or over designing - Help to discuss: What are some UI/UX design constraints for the current project? For example, what specific libraries should we try and stick with? What about styling and states? What can be changed, what can't, or ideally, wouldn't be changed right now? There's a risk of over or under-customizing a notebook.

**Feasibility Risks**

- Will we understand javascript and python within a scientific computing environment enough to contribute to the project?
- Are there technical particularities to deploying and testing in such an environment?
- Will we be able to implement all necessary functionality using the Leafmap framework as a starting point? Will we need to determine pieces that can be implemented before starting?

**Business Risks:**

- Can we differentiate ourselves enough from Leafmap?
- What's the cost of maintenance over time?

### Assumptions

- We can access end-users and request feedback
- The initial scope of work is focused on the re-design of the application
- It's possible to continue working on the project after the initial phase
- and are the main points of contact and not the IMPACT stakeholders
- We know the key workflows that need to be implemented
- Release Early, Release often - ideally, we would push new features for user testing every sprint.
- The target date for the initial onboarding of actual users by June 12, when we can get direct feedback from the User Working Group on existing and future features, usability, and other criteria.
- We want to release an early alpha version or have done some user validation before the onboarding date of users on June 12th
- ipyleafet is working in the ADE, and stac_ipyleafet can be installed as part of the Base Image all users start from.

### Milestones

- [Milestone 0: Kickoff - Week 0](https://github.com/MAAP-Project/stac_ipyleaflet/issues/40)
- [Milestone 1: Discovery Review - Week 2](https://github.com/MAAP-Project/stac_ipyleaflet/issues/41)
- [Milestone 2: Design review - Week 3](https://github.com/MAAP-Project/stac_ipyleaflet/issues/42)
- [Milestone 3: Sprint Review - Week 5](https://github.com/MAAP-Project/stac_ipyleaflet/issues/43)
- [Milestone 4: Future Scope Workshop - Week 6](https://github.com/MAAP-Project/stac_ipyleaflet/issues/44)
- [Milestone 5: Future SoW Review - Week 7](https://github.com/MAAP-Project/stac_ipyleaflet/issues/45)

### Epics

- [Discovery](https://github.com/MAAP-Project/stac_ipyleaflet/issues/31)
- [UX & UI Re-Design](https://github.com/MAAP-Project/stac_ipyleaflet/issues/32)
- [Implement Re-Design](https://github.com/MAAP-Project/stac_ipyleaflet/issues/33)
- [Testing & Deployment](https://github.com/MAAP-Project/stac_ipyleaflet/issues/36)
- [User Validation](https://github.com/MAAP-Project/stac_ipyleaflet/issues/35)
- [Future Work](https://github.com/MAAP-Project/stac_ipyleaflet/issues/34)

### Roadmap

You can visualize the current version of the roadmap in this [GitHub view](https://github.com/orgs/MAAP-Project/projects/1/views/2). It should be considered our current work hypothesis, not hard deadlines.

## Team Roles

The project team will consist of the following members:

- Developers - Ed Keeble, Emma Paz, Matt Robinson, Tammo Feldmann, Anthony Lukach
- UI/UX designer - Heidi Mok
- Product Owner - Zac Deziel

## Communication Plan

The following communication plan will be established to ensure that the project team stays informed and up-to-date:

- Partner Kickoff Meeting
- Daily asynchronous updates - Daily updates will be provided by the Orbaco team members as on other pod projects
- Milestone meetings - Discovery Review, Design Review, Sprint Review, Future Scope Workshop, Future SoW Review
- Slack channel - Communication will occur within the MAAP channel.
- Project management tool - GitHub will track the project's progress and document our work.

## Partner Kickoff Meeting

We will want an IMPACT PI Objective that covers this sub-project for April-June.

**Comment by zacharyDez** [comment]
## Notes from the Kickoff Milestone Meeting

- Think about who could go to the user working group on June 13-15th at University of Baltimore
- How can we establish a baseline for NPS; can we measure how satisfied they are with their current workflows? The baseline is 'choose your adventure' with a jupyter notebook. Hard to compare apples to oranges. It could be within the future SoW (between release 0.2 and 0.3). We should be intentional within the discovery phase. We are working with the assumption that people will adopt the tool for new notebooks when available might be difficult for older notebooks.
- The only constraint of the design is that it has to work within the notebook. There are no existing users right now. We want users to visualize COGs in the STAC catalog with as few steps as possible. Already possible through the Leafmap library. 
- Part of the discovery will be getting into the existing notebooks - Zac will act as an internal user.
- 0.1 release planned, Orbaco wants to bring it to 0.2

## Team Allocation

- @emmalu  and Ed will lead technical efforts
- @heidimok will lead design efforts
- @zacharyDez will lead future SoW
- @abarciauskas-bgse and @wildintellect will provide support when needed

## Action Items

- Schedule time with users (coordinate with @abarciauskas-bgse and @zacharyDez)
- Schedule the next kickoff meeting (@zacharyDez)
- Schedule time in SmartSheet (@zacharyDez)
- Discovery phase plan (@heidimok , @emmalu , @zacharyDez, Ed)
